File: test/session_add_extension.cc
#include <nan.h>
#include <libtorrent-rasterbar/extensions.h>

NAN_METHOD(CreateEmptyPlugin) {
  v8::Local<v8::Function> cons = v8::Local<v8::Function>::Cast(info[0]);
  v8::Local<v8::Object> plugin = Nan::NewInstance(cons).ToLocalChecked();
  libtorrent_rasterbar::Plugin* obj = Nan::ObjectWrap::Unwrap<libtorrent_rasterbar::Plugin>(plugin);
  obj->Reset(boost::shared_ptr<libtorrent::plugin>(new libtorrent::plugin()));

  info.GetReturnValue().Set(plugin);
}

NAN_MODULE_INIT(Init) {
  Nan::Export(target, "createEmptyPlugin", CreateEmptyPlugin);
}

NODE_MODULE(addon, Init)

File: test/session_addextension.js
'use strict'
const test = require('tape')

test('Session#addTorrent', (t) => {
  const libtorrent = require('../')
  const Session = libtorrent.Session

  t.test('pass UTMetadata', (t) => {
    const session = new Session()
    session.addExtension(libtorrent.extensions.UTMetadata)

    t.end()
  })

  t.test('pass Plugin', (t) => {
    const addon = require('./addon_test')
    const ext = addon.createEmptyPlugin(libtorrent.extensions.Plugin)

    const session = new Session()
    session.addExtension(ext)

    t.end()
  })

  t.end()
})
